src/containers/widget/index.js

```diff
@@ -17,76 +16,101 @@ import { WidgetWrapper } from "../../utils/widget-wrapper";
 
 const DEFAULT_API_BASE_URL = "https://area402.herokuapp.com";
 
-const Widget = ({
-  currency,
-  imageSrc,
-  showModal,
-  token,
-  apiBaseUrl,
-  screenName,
-  widgetTitle,
-  welcomeMessage,
-  paymentOptions,
-  enableEmailSubscription,
-}) => {
-  const [isModalOpen, setModalIsOpen] = useState(showModal);
-  const [invoiceDetails, setInvoiceDetails] = useState(null);
-  const [currentScreen, setCurrentScreen] = useState(screenName);
-  const [fetchingInvoiceState, setFetchingInvoiceState] = useState(false);
-
-  useEffect(() => {
-    return () => setDefaults();
-  }, []);
-
-  const setDefaults = () => {
-    setModalIsOpen(false);
-    setInvoiceDetails(null);
-    setCurrentScreen("welcome-screen");
+class Widget extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      invoiceDetails: null,
+      fetchingInvoiceState: false,
+      isModalOpen: props.showModal,
+      currentScreen: props.screenName,
+    };
+  }
+
+  componentWillUnmount() {
+    this.setDefaults();
+  }
+
+  setDefaults = () => {
```

Comment:
why are those not written as: 

```
setDefaults() {
 ...
}
```

Reply:
If we don't use arrow function syntax (`const foo = () => {...}`) then we have to manually bind the functions in the constructor which adds extra lines of code, so to avoid that we prefer using arrow functions

Reply:
because when we pass on the functions to some other screens `this` would no longer be the widget?

Reply:
It is still `widget`, but the scope is automatically binded when we use arrow functions, we don't need to do:

```js
constructor(props) {
  super(props)

  this.setDefaults = this.setDefaults.bind(this)
}
```